Simplenote/CrashLogging.swift

```diff
@@ -8,21 +8,32 @@ import AutomatticTracks
 @objc(CrashLogging)
 class CrashLoggingShim: NSObject {
 
-    private static var sharedCrashLogging: CrashLogging?
+    private let crashLogging: CrashLogging
```

Comment:
@jkmassel from your suggestion, I got the idea of fully removing this class and calling its methods directly in the application delegate.

That's not possible because `CrashLogging` is a Swift type that's not visible to the Objective-C `AppDelegate`.

This is an attempt to remove the `static` component while staying within the Objective-C constraints. I created a private instance of the Swift `CrashLogging` that this shim calls via its instance methods visible to the Objective-C code.

Reply:
Wondering if we couldn't make the main `CrashLogging` Swift type be `@objc`-compatible in the first place?

Reply:
Thank you for the suggestion. 🙇 

`CrashLogging` comes from Tracks, so updating it would require another round trip. That's not a huge issue, just a possible time sink that might prevent us from landing this in `release/2.7`.

I'd be reluctant to go down that path, though, because I fear a rabbit hole. @jkmassel you have  more experience with that code, so maybe you can  tell me if I'm being overly cautious?

Maybe we should update the app delegate to be Swift, instead 😛 I'm only half joking. Obviously that would be a bigger piece of work, but it might a good time investment to modernize the codebase?

Reply:
Thanks for looking at this @mokagio. @AliSoftware – we could certainly do that, though Simplenote is the only Objective-C (iOS and macOS) App Delegate (and ideally IMHO wouldn't own the Tracks `CrashLogging` class directly). 

FWIW it wouldn't be that hard to do, just need to conform to `NSObject`.